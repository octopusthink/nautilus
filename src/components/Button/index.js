import { css } from '@emotion/core';
import invariant from 'invariant';
import PropTypes from 'prop-types';
import React from 'react';

export const Button = ({
  children,
  minimal,
  navigation,
  primary,
  ...otherProps
}) => {
  invariant(
    (minimal === true && primary === true) === false,
    'Cannot use `minimal` and `primary` props together on a <Button> component.',
  );

  let Component = 'button';
  if (navigation === true) {
    Component = 'a';
  }

  return (
    // See: https://github.com/yannickcr/eslint-plugin-react/issues/1555
    // eslint-disable-next-line react/button-has-type
    <Component
      css={css`
        font-family: -apple-system;
        font-size: 17px;
        padding: 12px 32px;
        outline: none;
        text-decoration: none;
        border: 2px solid #c62e7f;
        color: #c62e7f;
        background: #fff;
        border-radius: 8px;
        margin: 0 16px 16px;
        line-height: 1.412;
        transition: box-shadow 200ms;

        &:active {
          border-color: #973367;
          color: #973367;
        }

        &:focus {
          box-shadow: 0 0 1px 4px #F5A3CE;
          outline: none;
        }

        &::-moz-focus-inner {
          border: 0;
        }

        &:hover {
          box-shadow: 0 2px 0 0 #973367;
        }

        ${primary &&
          css`
            background: #c62e7f;
            color: #fff;

            &:active {
              background: #973367;
              color: #fff;
            }
          `}

        ${minimal &&
          css`
            border-top: 0;
            border-left: 0;
            border-right: 0;
            border-radius: 0;
            padding-left: 0;
            padding-right: 0;
          `}

        ${navigation &&
          css`
            &::after {
              content: ' â†’';
              display: inline;
            }
          `}

          &:disabled {
            border-color: #9f9f9f;
            color: #9f9f9f;

            &:hover {
                box-shadow: 0 2px 0 0 #6C6C6C;
            }

            &:active {
              background: #6C6C6C;
            }

            &:focus {
              box-shadow: 0 0 1px 4px #D4D4D4;
            }

            &${primary &&
              css`
                background: #9f9f9f;
                color: #fff;
              `}

      `}
      {...otherProps}
    >
      {children}
    </Component>
  );
};

Button.defaultProps = {
  children: undefined,
  disabled: false,
  minimal: false,
  navigation: false,
  primary: false,
  type: 'button',
};

Button.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Disables the button; this applies a disabled style but **does not disable any event handlers for the button**. Your `onClick` handlers should check for `props.disabled` to modify their behaviour accordingly. */
  disabled: PropTypes.bool,
  /** Decrease the visual prominence of this button in the UI. */
  minimal: PropTypes.bool,
  /** Sometimes, you need a button that's actually a link. We got you! This outputs a `react-router-dom` `<Link>` tag that looks (and largely behaves) like a `<Button>`, but can used as navigation. Setting this to `true` enables `Link` properties; see: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md. */
  navigation: PropTypes.bool,
  /** Increase the visual prominence of this button in the UI. */
  primary: PropTypes.bool,
  /** HTML `type` attribute for the button. Defaults to `"button"`. */
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
};

export default Button;
