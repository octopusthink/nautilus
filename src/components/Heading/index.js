import { css } from '@emotion/core';
import invariant from 'invariant';
import PropTypes from 'prop-types';
import React from 'react';

import { useTheme } from 'themes';
import { headingLarge, headingMedium, headingSmall } from 'themes/mixins';

export const Heading = ({
  children,
  level,
  large,
  medium,
  small,
  ...otherProps
}) => {
  invariant(level, 'Heading component requires a level prop');

  const HeadingElement = `h${level}`;
  const theme = useTheme();

  return (
    <HeadingElement
      css={css`
        ${css(headingMedium(theme))};
        margin: 0 0 ${theme.spacing.margin.m};

        ${large &&
          css`
            ${css(headingLarge(theme))}
          `}

        ${small &&
          css`
            ${css(headingSmall(theme))}
          `}
      `}
      {...otherProps}
    >
      {children}
    </HeadingElement>
  );
};

export const HeadingLevels = [1, 2, 3, 4, 5, 6];

Heading.defaultProps = {
  children: undefined,
  level: 2,
  large: true,
  medium: false,
  small: false,
};

Heading.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Semantic hierarchy level of the `<h>` element in the markup (ex: `<h3>`) */
  level: PropTypes.oneOf(HeadingLevels).isRequired,
  /** Visual prominency of the heading. Small, medium, large. */
  large: PropTypes.bool,
  medium: PropTypes.bool,
  small: PropTypes.boolm,
};

export default Heading;
