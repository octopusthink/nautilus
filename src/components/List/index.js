import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { bodySmall, bodyMedium, bodyLarge } from 'themes/mixins';

export const List = ({
  children,
  large,
  small,
  inverse,
  dark,
  light,
  ...otherProps
}) => {
  return <ul {...otherProps}>{children}</ul>;
};

List.defaultProps = {
  children: undefined,
  large: false,
  small: false,
  inverse: false,
  dark: false,
  light: false,
};

List.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Increase the visual prominence of the paragraph. */
  large: PropTypes.bool,
  /** Decrease the visual prominence of the paragraph. */
  small: PropTypes.bool,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** Lighten text colour. */
  light: PropTypes.bool,
};

export default styled(List)(({ dark, inverse, large, light, small, theme }) => {
  return css`
      color: ${theme.colors.text.default};
      margin: 0 0 ${theme.spacing.margin.m};
      ${small && bodySmall(theme)};
      ${!small && !large && bodyMedium(theme)};
      ${large && bodyLarge(theme)};
      ${light &&
        css`
          color: ${theme.colors.text.light};
        `}

      ${dark &&
        css`
          color: ${theme.colors.text.dark};
        `}

      ${inverse &&
        css`
          color: ${theme.colors.text.inverse};

          ${light &&
            css`
              color: ${theme.colors.text.inverseLight};
            `}
          ${dark &&
            css`
              color: ${theme.colors.text.inverseDark};
            `}
        `}
    `;
});
