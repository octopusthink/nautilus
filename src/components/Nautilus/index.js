import { Global, css } from '@emotion/core';
import { ThemeProvider } from 'emotion-theming';
import PropTypes from 'prop-types';
import React from 'react';

import { nautilus as nautilusDefaultTheme } from 'themes';

export const Nautilus = ({ children, theme }) => {
  return (
    <ThemeProvider theme={theme}>
      <Global
        styles={css`
          html {
            /* Makes type rendering look crisper. */
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;

            /* Set base font size for rem units to inherit .*/
            font-size: 62.5%;

            /* Enable kerning and optional ligatures. */
            text-rendering: optimizeLegibility;

            /* Disable auto-enlargement of small text in Safari. */
            text-size-adjust: 100%;
          }
        `}
      />
      {children}
    </ThemeProvider>
  );
};

Nautilus.defaultProps = {
  children: undefined,
  theme: nautilusDefaultTheme,
};

Nautilus.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Theme object used to style this instance of Nautilus and its components. */
  theme: PropTypes.object,
};

export default Nautilus;
