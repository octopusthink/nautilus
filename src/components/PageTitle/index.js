import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { pageTitle } from 'themes/mixins';

export const PageTitle = ({ children, documentTitle, ...otherProps }) => {
  // const documentTitleToUse = documentTitle || children;

  return <h1 {...otherProps}>{children}</h1>;
};

PageTitle.defaultProps = {
  children: undefined,
  documentTitle: undefined,
};

PageTitle.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** String to pass to the `<DocumentTitle />` tag if the children of your `<PageTitle />` are more than just text content. */
  documentTitle: PropTypes.string,
};

export default styled(PageTitle)(({ theme }) => {
  return css`
    ${pageTitle(theme)}
  `;
});
