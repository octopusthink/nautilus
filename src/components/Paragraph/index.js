import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { textStyles } from 'themes/mixins';

export const Paragraph = ({
  children,
  large,
  small,
  inverse,
  dark,
  light,
  ...otherProps
}) => {
  return <p {...otherProps}>{children}</p>;
};

Paragraph.defaultProps = {
  children: undefined,
  large: false,
  small: false,
  inverse: false,
  dark: false,
  light: false,
};

Paragraph.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Increase the visual prominence of the paragraph. */
  large: PropTypes.bool,
  /** Decrease the visual prominence of the paragraph. */
  small: PropTypes.bool,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** Lighten text colour. */
  light: PropTypes.bool,
};

const StyledParagraph = styled(Paragraph)(textStyles);

export default StyledParagraph;
