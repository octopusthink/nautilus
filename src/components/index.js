export { default as Heading } from './Heading';
export { default as Paragraph } from './Paragraph';
