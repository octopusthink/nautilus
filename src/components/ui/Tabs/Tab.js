import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { toUnits } from 'styles';
import { useTheme } from 'themes';

export const Tab = (props) => {
  const { children, label, ...otherProps } = props;
  const theme = useTheme();

  return (
    <li role="presentation">
      <a
        css={css`
          padding: ${toUnits(theme.spacing.padding.small)};
          text-decoration: none;
          color: ${theme.colors.neutral.grey800};
        `}
        role="tab"
        href="#section1"
        id="tab1"
        aria-selected="true"
      >
        {label}
      </a>
    </li>
  );
};

export const styles = (props) => {
  const { theme } = props;

  return css`
    margin: 0 ${toUnits(theme.spacing.padding.small)} 0 0;
  `;
};

Tab.defaultProps = {
  children: undefined,
};

Tab.propTypes = {
  /** @ignore */
  children: PropTypes.node,
};

Tab.displayName = 'Tabs.Tab';

export default styled(Tab)(styles);
