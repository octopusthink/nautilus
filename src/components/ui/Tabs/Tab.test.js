import React from 'react';

import { muteConsole, render } from 'utils/testing';

import { Paragraph } from 'components';
import Tab from './Tab';

describe('Tabs.Tab', () => {
  it('should render nothing when no children are provided', () => {
    muteConsole({ times: 1, type: 'error' });
    const { container } = render(<Tab />);

    expect(container.firstChild).toBeNull();
  });

  it('should render a <section> tag', () => {
    const { container } = render(
      <Tab label="About">
        <Paragraph>Puppies are cute.</Paragraph>
      </Tab>,
    );

    expect(container.firstChild.tagName).toEqual('SECTION');
  });

  it('should render the label inside an h4', () => {
    const { container } = render(
      <Tab label="About">
        <Paragraph>Puppies are cute.</Paragraph>
      </Tab>,
    );

    expect(container.firstChild.children[0].tagName).toEqual('H4');
  });

  it('should render the children as-provided', () => {
    const { container } = render(
      <Tab label="About">
        <Paragraph>Puppies are cute.</Paragraph>
      </Tab>,
    );

    expect(container.firstChild.children[1].tagName).toEqual('P');
  });

  it('should render styles for a Tab', () => {
    const { container } = render(
      <Tab label="About">
        <Paragraph>Puppies are cute.</Paragraph>
      </Tab>,
    );

    expect(container).toMatchSnapshot();
  });
});
