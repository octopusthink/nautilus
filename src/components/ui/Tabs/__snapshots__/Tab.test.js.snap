// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs.Tab should not render styles when unstyled prop is set 1`] = `
.emotion-1 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

<div>
  <section
    class="emotion-0"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-1"
    >
      Puppies are cute.
    </p>
  </section>
</div>
`;

exports[`Tabs.Tab should render styles for a Tabs.Tab 1`] = `
.emotion-0 {
  display: none;
}

.emotion-0:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-1 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

<div>
  <section
    class="emotion-0"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-1"
    >
      Puppies are cute.
    </p>
  </section>
</div>
`;
